Doesn't build on GHC HEAD

Error:

```
Foundation/Collection/InnerFunctor.hs:13:13: error:
    • The default type signature for imap:
        forall (f :: * -> *) a.
        (Functor f, Element (f a) ~ a, f a ~ c) =>
        (a -> a) -> f a -> f a
      does not match its corresponding non-default type signature
    • When checking the class method:
        imap :: forall c.
                InnerFunctor c =>
                (Element c -> Element c) -> c -> c
      In the class declaration for ‘InnerFunctor’
   |
13 |     default imap :: (Functor f, Element (f a) ~ a, f a ~ c) => (a -> a) -> f a -> f a
   |             ^^^^
```

--- a/Foundation/Collection/InnerFunctor.hs
+++ b/Foundation/Collection/InnerFunctor.hs
@@ -12,7 +12,7 @@ import           Foundation.Array.Boxed (Array)
 -- | A monomorphic functor that maps the inner values to values of the same type
 class InnerFunctor c where
     imap :: (Element c -> Element c) -> c -> c
-    default imap :: (Functor f, Element (f a) ~ a, f a ~ c) => (a -> a) -> f a -> f a
+    default imap :: (Functor f, Element (f a) ~ a, f a ~ c) => (Element c -> Element c) -> c -> c
     imap = fmap
 
 instance InnerFunctor [a]

